File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.cs
using System;
using System.Linq;

public class Solution 
{
    public string[] solution(string myString) 
    {
        string[] str = myString.Split('x');
        Array.Sort(str);
        
        // Where메소드로 조건에 맞는 요소들만 필터링
        // !string.IsNullOrEmpty로 문자열이 비어있지 않고 null이 아닌 경우만 포함하도록 함
        return str.Where(w => !string.IsNullOrEmpty(w)).ToArray(); 
        
    }
}

File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(string myString) 
{
    vector<string> answer;
    string temp = "";
    for(int i=0; i<myString.length(); i++)
    {
        if(myString.compare(i,1,"x") == 0)
        {
            if(!temp.empty())
            {
                answer.push_back(temp);
                temp = "";
            }
        }
        else
        {
            temp += myString[i];
        }
    }
    if(!temp.empty())
    {
        answer.push_back(temp);
    }
    sort(answer.begin(), answer.end());
    return answer;
}